tools/prep/prepare_space_xml.py

```diff
@@ -235,6 +248,9 @@ def add_solvers(track, filter_benchmarks, select_benchmarks):
                       add_solvers_in_space(subspace, solvers)
             # remove spaces without solvers
             remove_spaces_without_solvers(space)
+            # remove non-competitive divisions
```

Comment:
Not needed (see above, we do this via `is_competitive`).

Reply:
> @aehyvari this is not unsat core specific, right? This would only extend the semantics of competitiveness of a division to what is stated in the rules. Which we should do, eventually. This year, fortunately, we don't have these cases, so maybe we can leave it for now. But let's discuss this offline.

Yes, this commit is generic, while the pull request itself addresses the issues I encountered while preparing for unsat-core track.

The script tools/prep/prepare_space_xml.py  has two implementations for the non-competitiveness selection, since I didn't remove your implementation, but simply added mine there.  This was because we implemented the thing twice without knowing of each others.

I'm going to defend my implementation here with these two arguments:

- a separate list of non-competitive divisions is better than the list of non-competitive solvers, since (1) this list is also useful for generating the benchmark tables for the website, and (2) the list of non-competitive divisions is a concept in the rules and the logic of the competition that is separated from the question whether a solver is competitive or not.  It is not hard to imagine cases where a division is declared non-competitive for reasons independent of which solvers participate in it.

- the list of non-competitive solvers should in any case not be given as an argument, since that information is already available in the csv and duplicating the information could cause confusion.

However, I wonder if there's a better way of doing the solver selection for unsat-core track and model verification track.  I think to answer this we should specify what is the role difference between this script and the selection script in https://github.com/SMT-COMP/smt-comp/blob/master/tools/selection/selection.py.   Can the filtering for unsat-core and model verification instances be moved to selection.py?